### PR TITLE
LL-2494

### PR DESCRIPTION
**Pull request by LFBarreto**
:atom:  delegation flow added using mocks

### Type

Feature

### Context

LL-2494

### Parts of the app affected / Test plan

Run the app using mock :atom: accounts and try out the delegation flow.
More details on the main issue

### CODE REVIEW

File: src/renderer/families/cosmos/DelegationFlowModal/index.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+class DelegationModal extends PureComponent<{ name: string }, State> {
+  state = INITIAL_STATE;
+
+  handleReset = () => this.setState({ ...INITIAL_STATE });
+
+  handleStepChange = (stepId: StepId) => this.setState({ stepId });
+
+  handleReset = () =>
```
Comment (JunichiSugiura) [inline src/renderer/families/cosmos/DelegationFlowModal/index.js:22]: Duplicated

File: src/renderer/families/cosmos/DelegationFlowModal/index.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    this.setState({
+      stepId: "castDelegations",
+    });
+
+  handleStepChange = (stepId: StepId) => this.setState({ stepId });
```
Comment (JunichiSugiura) [inline src/renderer/families/cosmos/DelegationFlowModal/index.js:27]: Duplicated

File: src/renderer/components/Delegation/ValidatorRow.js [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+  background-color: rgba(0, 0, 0, 0);
+  border: 1px solid transparent;
+  border-radius: 4px;
+  border-color: ${p => (p.notEnoughVotes ? p.theme.colors.pearl : p.theme.colors.palette.divider)};
+  &:disabled {
+    cursor: disabled;
+    color: ${p => p.theme.colors.palette.text.shade40};
+    background-color: ${p => p.theme.colors.palette.background.default};
+  }
+`;
```
Comment (gre) [inline src/renderer/components/Delegation/ValidatorRow.js:149]: I don't 100% understand why we always need to redefine input style again and again. Don't we have an Input component?

File: src/renderer/families/cosmos/DelegationFlowModal/fields/ValidatorsField.js [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+    unit,
+  );
+  const delegationsUsed = delegations.reduce((sum, v) => sum + formatValue(v.amount, unit), 0);
+  const delegationsSelected = delegations.length;
+
+  const max = Math.max(0, delegationsAvailable - delegationsUsed);
+
+  const onUpdateDelegation = useCallback(
+    (address, value) => {
+      const raw = value ? parseInt(value.replace(/[^0-9]/g, ""), 10) : 0;
```
Comment (gre) [inline src/renderer/families/cosmos/DelegationFlowModal/fields/ValidatorsField.js:136]: It feels these kind of boilerplate are rewritten everywhere again. we have parsers in live-common 🤔

File: src/renderer/families/cosmos/DelegationFlowModal/fields/ValidatorsField.js [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+
+      const amount =
+        raw <= 0 || delegationsSelected > MAX_VOTES
+          ? BigNumber(0)
+          : BigNumber(raw).multipliedBy(10 ** unit.magnitude);
```
Comment (gre) [inline src/renderer/families/cosmos/DelegationFlowModal/fields/ValidatorsField.js:141]: raw should be already in correct magnitude in the first place if we use a AmountInput component. (we should try to use it / adapt from it what needs to be adapted)

**Pull request merged**